### ISSUE TITLE
System Monitor - showing MySQL and not MariaDB

### ISSUE THREAD

**Issue by shoulders** [virtualmin/virtualmin-gpl#798]
|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Ubuntu Linux 22.04.4     |
|  Webmin version      |     2.111      |
|  Usermin version     |     2.010     |
|  Virtualmin version  |     7.10.0       |
|  Theme version       |     21.10      |
|  Package updates     |     All installed packages are up to date   |

### The issue

In system monitors it is using `MySQL Database Server` and not `MariaDB Database Server`

![image](https://github.com/virtualmin/virtualmin-gpl/assets/319997/1e849c6c-4001-42a1-9115-c549bfd93b56)

`server status` shows correctly

![image](https://github.com/virtualmin/virtualmin-gpl/assets/319997/520fe161-addf-45c6-a8b0-188a98010172)

### proposed solution

`MySQL Database Server` --> `MariaDB Database Server` as fitting the installed technology

### Additional

`Webmin --> tools --> System and Server Status`

![image](https://github.com/virtualmin/virtualmin-gpl/assets/319997/895274b1-993f-4397-9391-ee62a5dbedb9)

I can then edit `MySQL Database Server`

![image](https://github.com/virtualmin/virtualmin-gpl/assets/319997/db6e2e57-4153-4427-adf4-952754fcd580)

- The description is manually editable, and I can do that if it is the way.
- `MySQL Database Server` is not user editable. Now I understand that Virtualmin can utilise MySQL aswell

**Comment by iliajie** [comment]
Hello,

This is a good suggestion, and it could easily be fixed from the `list_services` sub. Though, I'm not sure if it's a clean solution.

We'll wait for Jamie's comment on it!

**Comment by jcameron** [comment]
Yes, we should use "MariaDB" instead of "MySQL" in the default monitor name ... I'll fix this.

**Comment by iliajie** [comment]
> Yes, we should use "MariaDB" instead of "MySQL" in the default monitor name ... I'll fix this.

I've checked the latest patches and it would work just fine for new installations. However, existing installations don't get `services/mysql.serv` updated and therefore the label fix doesn't get triggered as `$serv->{'default'}` never there.

Should we also address that?

**Comment by iliajie** [comment]
Also, the monitor name isn't fixed in **Tools ⇾ System and Server Status: Edit Monitor** page.

**Comment by jcameron** [comment]
I've checked in another fix for that..

**Comment by iliajie** [comment]
Thanks!

Though it doesn't update **Description** field:

![image](https://github.com/virtualmin/virtualmin-gpl/assets/4426533/6a1e061c-e9a1-4deb-9bda-f185a609aaee)

Also, I think we should create a specialized and dedicated subroutine in the MySQL library to handle this type of label conversion. I don't like having callers reimplement this logic all the times!

**Comment by iliajie** [comment]
But it's confusing! A user clicks a MariaDB monitor and gets MySQL?

**Comment by iliajie** [comment]
Jamie, I just had a thought. Why don't we create a sub in Webmin main lib file like `module_variant`, then MySQL lib will cache a file in `/etc/webmin/mysql/variant` , and all other modules will just call `module_variant('mysql')` to get a variant from that cache file.

What do you think?

**Comment by jcameron** [comment]
> But it's confusing! A user clicks a MariaDB monitor and gets MySQL?

This will be corrected for new installs when the default monitors are created.

**Comment by iliajie** [comment]
> This will be corrected for new installs when the default monitors are created.

This is fine!

Though, at what point in time is a new default MySQL monitor created? What triggers its creation?

**Comment by jcameron** [comment]
First time the module is used

**Comment by jcameron** [comment]
The status module

**Comment by iliajie** [comment]
What if MariaDB server would be installed later?

**Comment by jcameron** [comment]
Sorry, in that case you're out of luck..

**Comment by iliajie** [comment]
LoL. 😅 Why not address it in more systematic way?

**Comment by jcameron** [comment]
Because the default set of monitors is created by copying templates, after which their descriptions are user-editable and so it wouldn't make sense for Webmin to programatically edit them.

**Comment by iliajie** [comment]
Okay, then let's not create default monitor until module (i.e. MariaDB or MySQL package) is installed?

**Comment by jcameron** [comment]
I suppose that could be done, but honestly it feels like we're overcomplicating things at this point.

**Comment by iliajie** [comment]
Even though I agree it's minor, I just wanted to have it done precise.

**Comment by iliajie** [comment]
I think we should create a special sub in MySQL lib, which would read cached details about database version and variant. As database server 99% never switched (i.e. MariaDB ↔ MySQL); we shouldn't reading it every time.

**Comment by jcameron** [comment]
There's a variable $mysql::mysql_version which can be used this for already

**Comment by iliajie** [comment]
> There's a variable $mysql::mysql_version which can be used this for already

Great! I've checked the code and you're right! We already cache MySQL/MariaDB version.

So, then I think we can just leave it as it is?
